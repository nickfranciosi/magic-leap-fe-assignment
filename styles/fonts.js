const sansSerifFallback = 'Helvetica, arial, sans-serif';
const monoSpaceFallback = 'monospace';

const fontFamilyUtil = (font, fallBack = sansSerifFallback) =>
  `'${font}', ${fallBack}`;

const fontFamilies = {
  primary: fontFamilyUtil('Lato'),
  secondary: fontFamilyUtil('Raleway'),
  accent: fontFamilyUtil('Space Mono', monoSpaceFallback)
};

const sizes = {
  base: '16px',
  medium: '24px',
  header: '32px',
  detail: '11px'
};

export default {
  sizes,
  fontFamilies
};
