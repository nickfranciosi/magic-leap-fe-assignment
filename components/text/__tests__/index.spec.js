/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import MainTitle from '../mainTitle';
import Message from '../message';
import SubTitle from '../subTitle';
import fonts from '../../../styles/fonts';

describe('MainTitle', () => {
  test('it should render', () => {
    const tree = renderer.create(<MainTitle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it should have a larger leading', () => {
    const tree = renderer.create(<MainTitle />).toJSON();
    expect(tree).toHaveStyleRule('letter-spacing', '2.5px');
  });
});

describe('Message', () => {
  test('it should render', () => {
    const tree = renderer.create(<Message />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('SubTitle', () => {
  test('it should render', () => {
    const tree = renderer.create(<SubTitle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it should have a header font size', () => {
    const tree = renderer.create(<SubTitle />).toJSON();
    expect(tree).toHaveStyleRule('font-size', fonts.sizes.header);
  });
});
