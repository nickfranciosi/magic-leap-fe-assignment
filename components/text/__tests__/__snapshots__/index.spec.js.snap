// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainTitle it should render 1`] = `
.c0 {
  color: #ddd;
  font-family: 'Lato',Helvetica,arial,sans-serif;
  font-size: 16px;
  -webkit-letter-spacing: 2.5px;
  -moz-letter-spacing: 2.5px;
  -ms-letter-spacing: 2.5px;
  letter-spacing: 2.5px;
  text-transform: uppercase;
  -webkit-transition: color 350ms ease;
  transition: color 350ms ease;
  cursor: pointer;
}

.c0:hover {
  color: #fff;
}

<h1
  className="c0"
/>
`;

exports[`Message it should render 1`] = `
.c0 {
  font-size: 40px;
  font-family: 'Lato',Helvetica,arial,sans-serif;
  color: #fff;
  margin-bottom: 16px;
}

<h2
  className="c0"
/>
`;

exports[`SubTitle it should render 1`] = `
.c0 {
  font-size: 32px;
  color: #fff;
  font-family: 'Raleway',Helvetica,arial,sans-serif;
}

<h2
  className="c0"
/>
`;
